src/Shouldly/Internals/SourceCodeTextGetter.cs

```diff
@@ -58,7 +58,7 @@ void ParseStackTrace(StackTrace? trace)
                 // walk up stack trace until we get to the calling method might have to be updated regularly as we find more
                 // scenarios. Alternately, it could be replaced with a more robust implementation.
                 while (currentFrame.GetMethod().DeclaringType == null ||
-                        (currentFrame.GetMethod().DeclaringType.FullName?.StartsWith("System.Dynamic") ?? false))
+                        (currentFrame.GetMethod().DeclaringType.FullName?.StartsWith("System.Dynamic", StringComparison.Ordinal) ?? false))
```

Comment:
In two separate instances, StartsWith/EndsWith with the default CurrentCulture comparison in NUnit caused such bad performance for Jon Skeet that he reported it. For NUnit I'm proposing adding Fxcop Analyzers, but I'd recommend at least doing a one-time sweep.

Reply:
thanks. i had no idea. is this still a problem in the current netcore?

Reply:
I'm not sure, but I do know that the semantics issue is the same and there is still perf overhead associated with doing linguistic comparisons instead of binary.

Reply:
See https://github.com/nunit/nunit/pull/3583 for Jon's comment and links elsewhere

Reply:
The second one was just last month, perf hit was 5-6x

Reply:
perhaps i should re-purpose this code https://github.com/Fody/Caseless to make `StartsWith("System.Dynamic")` compile to  `StartsWith("System.Dynamic", StringComparison.Ordinal)` :)

Reply:
That sounds cool, but there's already so much confusion around these APIs, idk about adding contextual subtleties...